src/typings/prop-types.d.ts

```diff
@@ -0,0 +1,7 @@
+declare module 'prop-types' {
+    import React = require('react');
+
+    export const bool: {
+        isRequired: React.Validator<any>
```

Comment:
I've exported bool only because it's the only PropType used in the project for now but you're right, I'll use what you mentioned in the EDIT.